### PR TITLE
Migration docs

### PR DESCRIPTION
**Pull request by JohnC-80**
# Purpose
Adding some informational docs here that discuss managing breaking changes and versioning. 

## Migration paths
A guide for making changes due to upgrading stripes components. Ideally create a permanent document that we might be able to link to from deprecation console warnings or any other documentation to facilitate the upgrade process from any previous version.

There's probably already more that we can add to the migration document. Just floating this idea...

### Rendered markdown links for convenience...
[Change mangement](https://github.com/folio-org/stripes-components/blob/migration-docs/CHANGEMANAGEMENT.md)
[Migration paths](https://github.com/folio-org/stripes-components/blob/migration-docs/MIGRATIONPATHS.md)

### CODE REVIEW

File: CHANGEMANAGEMENT.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Breaking Changes and Deprecations
+Change happens. We learn more, we find better ways to do things. We clean house and beautify API's, improve user experience, accessibility, and developer experience. 
+Through the RFC(Request for comment) process, modifications to the component library are proposed and agreed upon by the community of FOLIO developers. The outcome of this process is an iterative plan to accomplish the proposed goal. The changes are weighed against NFR's - not limited to maintainability, platform stability, testability, extent of migration effort - with breaking change as final resort.
```
Comment (cherewaty) [inline CHANGEMANAGEMENT.md]: would spell out "non-functional requirements"; took me a minute to process that acronym

File: CHANGEMANAGEMENT.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+## Break as softly as possible.
+Through adequate release planning and up-front migration development, we aim to facilitate platform stability and reduce development stress, providing adequate timelines for migration to new code.
+## Semver
+We use the Semver versioning system where changes to the Major number(the first one) represent a present backwards incompatibility.
+Some examples of how we determine version number changes:
+* **Patch Version -** fully compatible bug fixes, minor improvements. ex: `3.0.9` to `3.0.10`.
+* **Minor Version -** new props, components, capabilities. ex: `3.0.10` to `3.1.0`.
+* **Major Version -** an incompatible change is present in the system. Removal of a component or component props with no co-existing replacement. ex: `3.1.0` to `4.0.0`.
+## Console Warnings
+Whenever possible, the release of breaking changes is proceeded by console warnings. These may appear months prior to the actual breaking change, depending on the extent of migration effort.
```
Comment (zburke) [inline CHANGEMANAGEMENT.md]: s/proceeded/preceded/

File: CHANGEMANAGEMENT.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Breaking Changes and Deprecations
+Change happens. We learn more, we find better ways to do things. We clean house and beautify API's, improve user experience, accessibility, and developer experience. 
+Through the RFC(Request for comment) process, modifications to the component library are proposed and agreed upon by the community of FOLIO developers. The outcome of this process is an iterative plan to accomplish the proposed goal. The changes are weighed against NFR's - not limited to maintainability, platform stability, testability, extent of migration effort - with breaking change as final resort.
+## Break as softly as possible.
+Through adequate release planning and up-front migration development, we aim to facilitate platform stability and reduce development stress, providing adequate timelines for migration to new code.
+## Semver
+We use the Semver versioning system where changes to the Major number(the first one) represent a present backwards incompatibility.
```
Comment (zburke) [inline CHANGEMANAGEMENT.md]:
```
number (the...
      ^
      |
```

File: CHANGEMANAGEMENT.md (hunk 1 above)
Comment (zburke) [reply CHANGEMANAGEMENT.md]: missing space after RFC, as long as you're tweaking that line.

File: MIGRATIONPATHS.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Stripes-Components Migration Paths
+## 3.x to 4.x
+Upcoming release.
+[Some Components and utilities have moved](#some-components-and-utilities-have-moved)
+### Some Components and utilities have moved
+We've provided console warnings for these items. If you've used these in your module with a `stripes-components` path, you'll simply have to update your import path:
+
+Component/Util | New path
+-- | --
+`<AddressList>` | `@folio/stripes-smart-components/lib/AddressFieldGroup/AddressList`
```
Comment (zburke) [inline MIGRATIONPATHS.md]: @cherewaty configured exports on `stripes-smart-components/index.js` so we can directly import from `@folio/stripes-smart-components` without needing the additional path. This is true for all but `EmbeddedAddressForm` -- was that deliberate?

File: CHANGEMANAGEMENT.md (hunk 3 above)
Comment (JohnC-80) [reply CHANGEMANAGEMENT.md]: whoops... thanks! (insert hilarious comment about `mdlint`-ishness)

**Comment by JohnC-80** [comment]
Thanks for the feedback, guys! 
@cherewaty we can update the path for `<EmbeddedAddressForm>` once we get that reconciled.

Not sure how to best format the 'path moves' table - markdown is surprisingly finicky about block elements within tables.

**Pull request merged**